### ISSUE TITLE
Use ReactJS for some UI elements

### ISSUE THREAD

**Issue by quincylvania** [openstreetmap/iD#6540]
While D3 is great for displaying loads of data elements like we need for iD's map, it can be cumbersome for newcomers to learn. Simply adding a button or label can be challenging and error-prone for new contributors unfamiliar with D3.

We should look into using [React](https://reactjs.org) for most UI elements instead of D3. React is more widely used than D3 and could make certain UI code easier to write and understand.

**Comment by BjornRasmussen** [comment]
Question: will both React and D3 have to be loaded every time users launch iD instead of just D3?

**Comment by quincylvania** [comment]
Yes, we will likely keep D3 around for map rendering for awhile.

**Comment by til-schneider** [comment]
I would like to contribute this.

Is there a (small) part of the UI which would be a good candidate to be turned into React as an example?

If you (@BjornRasmussen) have concerns React could be a too heavy dependency, we could use [preact](https://www.npmjs.com/package/preact) instead. preact is API-compatible to React (so we could switch easily to React later) and only needs 3 kB (instead of [109 kb (34.8 kb gzipped) for react + react-dom](https://reactjs.org/blog/2017/09/26/react-v16.0.html#reduced-file-size)). The main difference is that preact works directly on the DOM instead of using a shadow DOM. So it's slightly slower for very large UIs. But IMHO 34.8 kb gzipped is OK.

**Comment by til-schneider** [comment]
I just opened a pull request adding support for ReactJS. See #7861.
